src/main/java/com/griddynamics/mappers/ProductMapper.java

```diff
@@ -0,0 +1,137 @@
+package com.griddynamics.mappers;
+
+import com.griddynamics.dto.CategoryDTO;
+import com.griddynamics.dto.ProductDTO;
+import com.griddynamics.entities.Category;
+import com.griddynamics.entities.Product;
+import com.griddynamics.exceptions.MappingException;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Component
+public class ProductMapper extends EntityMapper<Product, ProductDTO> {
+
+    @Override
+    public ProductDTO mapDTO(Product product, ProductDTO productDTO) {
+        if (product == null || productDTO == null) {
+            throw new MappingException("One of arguments is null.");
+        }
+        productDTO.setId(product.getId());
+        productDTO.setName(product.getName());
+        productDTO.setPrice(product.getPrice());
+        productDTO.setDescription(product.getDescription());
+        productDTO.setBrand(product.getBrand());
+        productDTO.setImage(product.getImage());
+
+        List<CategoryDTO> categoryDTOList = new ArrayList<>();
+
+        productDTO.setCategoryDTOList(categoryDTOList);
+
+        List<Category> categoryList = product.getCategory();
+
+        CategoryMapper categoryMapper = new CategoryMapper();
```

Comment:
i dont like this approach. in that case for each call we create new instance of mapper